### PR TITLE
Changes to remove all possible hard coded query parameters from the request mapping url

### PR DESCRIPTION
**Pull request by kunal-pmj**
Changes to remove all possible hard coded query parameters from the request mapping url and move them to Parameters section

### CODE REVIEW

File: vmsgen.py [hunk 1]
```diff
@@ -619,12 +624,76 @@ def create_unique_op_ids(path_dict):
...
+    z.update(y)    # modifies z with y's keys and values & returns None
+    return z
+
+
+def remove_query_params(path_dict):
+    """
+    Swagger/Open API specification prohibits describing query parameter as part of the the request mapping url.
+    This method attempts to remove the Query Parameters from the Path and have them in the parameter section.
+     in the Swagger.
+    APIs which result in duplicate APIS on removing the query parameters from the path will be left as is.
```
Comment (sreeshas) [inline vmsgen.py]: Its not clear what happens when two operations have same path and same http verb. Can you improve the description?

Comment (kunal-pmj) [reply vmsgen.py]: The method first checks If removing a query parameter would result in 'absolute duplicate'. i.e same path and HTTP Verb. In that case the query param will NOT be removed from the path.

File: vmsgen.py [hunk 2]
```diff
@@ -619,12 +624,76 @@ def create_unique_op_ids(path_dict):
...
+
+    On removing the Query Parameter from the Path following 4 scenarios are possible:
+
+    1. The Path and the Operation Types are same as one of the existing Path:
+        Handling Such APIs is Out of Scope of this method. These APIs will appear
+         in the new Swagger definition without any change.
+        Example :
+                /com/vmware/cis/session?~action=get : [POST]
+                /com/vmware/cis/session : [POST, DELETE]
+    2. The Paths are duplicate but the operations are Unique:
```
[the remark this answers is not in the document]
Comment (kunal-pmj) [reply vmsgen.py]: The uniqueness is determined by combination of path and HTTP Operation. 
I will do one more check in with the updated comment and examples

File: vmsgen.py [hunk 3]
```diff
@@ -632,28 +632,47 @@ def merge_dictionaries(x, y):
...
+    Swagger/Open API specification prohibits appending query parameter to the request mapping path.
 
-    On removing the Query Parameter from the Path following 4 scenarios are possible:
+    Duplicate paths in Open API :
+        Since request mapping paths are keys in the Open Api JSON, there is no scope of duplicate request mapping paths
 
-    1. The Path and the Operation Types are same as one of the existing Path:
-        Handling Such APIs is Out of Scope of this method. These APIs will appear
-         in the new Swagger definition without any change.
+    Partial Duplicates in Open API: APIs which have same request mapping paths hut different HTTP Operations.
```
Comment (sreeshas) [inline vmsgen.py]: nit: "but"

File: vmsgen.py [hunk 4]
```diff
@@ -1041,6 +1119,8 @@ def get_input_params():
     GENERATE_UNIQUE_OP_IDS = args.unique_operation_ids
     global TAG_SEPARATOR
     TAG_SEPARATOR = args.tag_separator
+    global REMOVE_QUERY_PARAMS_FROM_PATH
```
Comment (sreeshas) [inline vmsgen.py]: I think this feature flag is unnecessary.  by default we can remove query params from path where applicable.

**Comment by kunal-pmj** [comment]
Hi @sreeshas 

I have made the changes as per your review comments.

Thanks
Kunal

**Pull request merged**